geoips/interfaces/base.py

```diff
@@ -460,8 +412,59 @@ def get_plugin(self, name):
         ProductsInterface which uses a tuple containing 'source_name' and
         'name'.
         """
+        if not self._unvalidated_plugins:
+            raise PluginRegistryError(
+                "Plugin registry not found, please run 'create_plugin_registry'"
+            )
         try:
```

Comment:
This try/except block is really big. It runs the risk of capturing errors other than the error we expect to catch. Is there a way to simplify this to move as much as possible outside of the try/except?

Reply:
I don't think this actually needs a try/except block. We know the structure of registered_plugins.yaml, control how it's built, so we shouldn't encounter a KeyError here, based on our own structural design.

Reply:
Are we going to just remove the try/except here?

Reply:
Let me remove it and run the full_test.sh script again.

Reply:
I added two smaller ones with applied Error Messages to get this to work. my_layered_test_script.sh (found in template_fusion_plugin), failed without a try/except statement, so it was required all in all. Smaller the better though! Feel free to resolve if you like this.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/240/files/#r1373527621
https://github.com/NRLMMD-GEOIPS/geoips/pull/240/files/#r1373528157

Those 2 try/excepts added to replace the one large try/except. Looks good

Reply:
Agreed, this looks good! Resolving.